central/deployment/store/cache/cache.go

```diff
@@ -140,6 +143,13 @@ func (c *cacheImpl) Walk(_ context.Context, fn func(deployment *storage.Deployme
 	return nil
 }
 
+func (c *cacheImpl) WalkByQuery(ctx context.Context, query *v1.Query, fn func(deployment *storage.Deployment) error) error {
+	if proto.Equal(query, search.EmptyQuery()) {
```

Comment:
Is this necessary?  Shouldn't `WalkByQuery` with an empty query be functionally the same as a `Walk`?

Reply:
This is nice because if empty then it'll hit the cache instead of the store itself